addons/purchase_stock/models/stock_rule.py

```diff
@@ -44,6 +44,17 @@ def _onchange_action(self):
         if self.action == 'buy':
             self.location_src_id = False
 
+    def _get_po_line_grouping_key(self, po_line):
+        """Construct a grouping key for purchase order lines based on the product_id.
+        This key will be used to group order lines by product.
```

Comment:
Please indicate that this method exists to be overridden, otherwise someone could decide to remove it later.